src/generic/stage2_build_tape.h

```diff
@@ -308,45 +393,56 @@ unified_machine(const uint8_t *buf, size_t len, ParsedJson &pj) {
     /* we found an array inside an object, so we need to increment the depth
      */
     depth++;
-    if (depth >= pj.depth_capacity) {
-      goto fail;
+    if(depth < pj.depth_capacity) {
```

Comment:
Because this is a forward jump, I would expect this to hurt branch prediction--at least according to Intel, forward conditional branches default to false.

Reply:
You might be right.

Reply:
Empirically, reversing it makes things worse, so this change was (slightly) beneficial.

Reply:
Huh. One day I'll understand branch prediction :) Mind including a note that it's empirically better? Since it doesn't go with our intuitions we should leave a note for future generations.

Reply:
I am not sure I want to make claims regarding optimality of this code at this time. What would the purpose be? To discourage you and others from trying to improve it? That would require me to know, with high certainty, that it is a dead-end. I do not know any such thing. In fact, I suspect that this is code is nowhere near optimal. I want to encourage further optimization efforts!!!

Misleading labels telling people that this code is faster this way is wrong, I think. I am working empirically and running lots of tests, but I am not at all confident.

Note that we have performance regression... it is now tiny, but I don't quite know where it comes from.